pkg/integrations/elasticsearch_exporter/elasticsearch_exporter.go

```diff
@@ -66,6 +70,21 @@ type Config struct {
 	ExportDataStreams bool `yaml:"data_stream,omitempty"`
 	// Export stats for Snapshot Lifecycle Management
 	ExportSLM bool `yaml:"slm,omitempty"`
+	// BasicAuth block allows secure connection with Elasticsearch cluster via Basic-Auth
+	BasicAuth *commoncfg.BasicAuth `yaml:"basic_auth,omitempty"`
```

Comment:
given that this also touches the integration implementation itself, we should also update the static configs doc https://grafana.com/docs/agent/latest/static/configuration/integrations/elasticsearch-exporter-config/

Reply:
Also, I think we should use in here the prometheus `BasicAuth` type, which is yaml serializable, in contrast with the river definition of it. If not, someone trying to use this feature in a static config agent won't be able to.

See that the river `BasicAuth` type has a `Convert` method that translates it to the prometheus type in [here](https://github.com/grafana/agent/blob/a1033f76ebc0b02b01ff551b81a2fa11254f1d2f/component/common/config/types.go#L136-L146).

@rfratto wdyt?

Reply:
Hi there, thanks for the review! Makes sense to continue supporting the static config :D. I've made amends in latest commit. PTAL if you have time.